dcos_launch/util.py

```diff
@@ -183,6 +186,26 @@ def generate_rsa_keypair(key_size=2048):
     return privkey_pem, pubkey_pem
 
 
+def run_subprocess(cmd: str, cwd: str, stream=True) -> str:
```

Comment:
I don't think you need this method at all. Check out this https://docs.python.org/3/library/subprocess.html#subprocess.check_output

It seems the only thing unique here is that you can stream this to log and capture it for output simultaneously. Do you really need that? Can you not just capture and then print?

Usually streaming is good for giant piles of logs from a long running process and capturing is for things you want to parse (simple and contained usually)

`run` and `call` may also be useful choices. You should probably not use Popen if you will immediately block and wait for the process to finish after it was started as the other commands do the error checking and take almost all of the same keyword args

Anyway, if you did need to absolutely stream and capture, I would try to make that a part of this method description. E.G. `tee_process` because you are kinda doing what `tee` does

Reply:
I do need streaming for create() because otherwise you don't know how things are progressing for 10 minutes. I need output for describe. So I made a generic method, but you're totally right, I don't need both at the same time. I'll use both check_call and check_output.